nixos/modules/services/x11/clight.nix

```diff
@@ -0,0 +1,149 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.clight;
+  clightConf = pkgs.writeText "clight.conf" cfg.extraConfig;
+in {
+  options.services.clight = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable clight or not.
+      '';
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.clight;
+      defaultText = "pkgs.clight";
+      description = ''
+        Clight derivation to use.
+      '';
+    };
+
+    clightd = mkOption {
+      type = types.package;
+      default = pkgs.clightd;
+      defaultText = "pkgs.clightd";
+      description = ''
+        Clightd derivation to use.
+      '';
+    };
+
+    extraConfig = mkOption {
+      type = types.str;
+      default = "";
+      description = ''
+        Additional configuration to pass to clight.conf.
+      '';
+    };
+
+    extraOptions = mkOption {
+      type = types.listOf types.str;
+      default = [];
+      description = ''
+        Additional command-line arguments to pass to
+        <command>clight</command>.
+      '';
+    };
+
+    latitude = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Your current latitude, between
+        <literal>-90.0</literal> and <literal>90.0</literal>. Must be provided
+        along with longitude.
+      '';
+    };
+
+    longitude = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Your current longitude, between
+        between <literal>-180.0</literal> and <literal>180.0</literal>. Must be
+        provided along with latitude.
+      '';
+    };
+
+    temperature = {
+      day = mkOption {
+        type = types.int;
+        default = 5500;
+        description = ''
+          Colour temperature to use during the day, between
+          <literal>1000</literal> and <literal>25000</literal> K.
+        '';
+      };
+      night = mkOption {
+        type = types.int;
+        default = 3700;
+        description = ''
+          Colour temperature to use at night, between
+          <literal>1000</literal> and <literal>25000</literal> K.
+        '';
+      };
+    };
+  };
+
+  config = let
+    manualLocation = cfg.latitude != null && cfg.longitude != null;
```

Comment:
An assertion for either both or none of these being set would be nice.

Actually this is a good reason for abstracting these into a `location.nix` module, because such assertions are already present in the redshift module, along with pretty much the same manual/automatic logic.